thrift/server.go

```diff
@@ -185,6 +186,17 @@ func (s *Server) handle(origCtx context.Context, handler handler, method string,
 		return err
 	}
 
+	// Buffer the response write so we can send a system error to the client.
+	// It's too late to send a system error by the time we write Arg3.
+	var respStruct bytes.Buffer // XXX should we make a buffer pool?
+	wp = getProtocolWriter(&respStruct)
+	if err := resp.Write(wp.protocol); err != nil {
+		call.Response().SendSystemError(err)
```

Comment:
This can return an err at which point it should be a tableflip

Reply:
Same goes with writing out the buffer, eg the socket died. Plenty of tableflip failure modes that will at least get a server side error.